component/core/src/main/java/org/exoplatform/social/core/listeners/GroupSynchronizationSocialProfileListener.java

```
@@ -0,0 +1,178 @@
+      try {
+        Group newPropertyValueGroup = getOrCreateGroup(propertyValue, newPropertyNameGroup);
+        addUserToGroup(newPropertyValueGroup, user);
+      } catch (Exception e) {
+        LOG.error("Error while adding property value group {} under property Group {}",propertyValue,newPropertyNameGroup!=null?newPropertyNameGroup.getId():"/", e);
+      }
+    } catch (Exception e) {
+      LOG.error("Error while adding property group {} under profile Group {}",propertyName,profileGroup!=null?profileGroup.getId():"/", e);
+    }
+  }
+
+  private Group getGroup(String groupId) {
+    try {
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      return groupHandler.findGroupById(groupId);
+    } catch (Exception e) {
+      LOG.warn("Group with Id: {} not found", groupId, e);
+      return null;
+    }
+  }
+
+  private Group getOrCreateGroup(String groupName, Group parentGroup) throws Exception{
+      Group group = getGroup(buildGroupId(parentGroup, groupName));
+      if (group != null) {
+        return group;
+      }
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      Group newGroup = groupHandler.createGroupInstance();
+      newGroup.setGroupName(groupName.toLowerCase());
+      newGroup.setLabel(StringUtils.capitalize(groupName));
+      newGroup.setDescription(groupName + " group");
+      groupHandler.addChild(parentGroup, newGroup, true);
+      return getGroup(buildGroupId(parentGroup, groupName));
+  }
+
+  private void addUserToGroup(Group group, User user) {
+    if (group == null) {
+      return;
+    }
+    try {
+      MembershipType membershipType = organizationService.getMembershipTypeHandler().findMembershipType(MEMBER);
+      organizationService.getMembershipHandler().linkMembership(user, group, membershipType, true);
+    } catch (Exception e) {
+      LOG.error("Error while adding user {} to Group {}",user.getUserName(),group.getId(), e);
```

to throw he exception instead of logging here to stop the processing in top layer method.

---

I'll throw the exception, but need to conserve the log to know if the problem comes from adding the group or the user to the group in the caller method